negotiator.go

```diff
@@ -46,16 +46,30 @@ func (l Negotiator) RoundTrip(req *http.Request) (res *http.Response, err error)
 		return rt.RoundTrip(req)
 	}
 	reqauthBasic := reqauth.Basic()
-	// Save request body
-	body := bytes.Buffer{}
+	// We need to buffer or seek the request body to handle authentication challenges
+	// that require resending the body multiple times during the NTLM handshake.
+	var body io.ReadSeeker
+	var bodyStartPos int64
 	if req.Body != nil {
-		_, err = body.ReadFrom(req.Body)
-		if err != nil {
-			return nil, err
+		// Check if body is already seekable to avoid buffering large bodies
+		if seeker, ok := req.Body.(io.ReadSeeker); ok {
+			body = seeker
```

Comment:
the `http.RoundTripper` interface says that "RoundTrip must always close the body, including on errors,".  Should we add a `defer req.Body.Close()` here?